internal/config/envconfig/load_test.go

```diff
@@ -0,0 +1,69 @@
+package envconfig_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/git-town/git-town/v19/internal/config/configdomain"
+	"github.com/git-town/git-town/v19/internal/config/envconfig"
+	"github.com/git-town/git-town/v19/pkg/prelude"
+	"github.com/shoenig/test/must"
+)
+
+func TestLoad(t *testing.T) {
+	const text = "my-github-token"
+	const alternateText = "my-github-auth-token"
+
+	t.Run("loads from GITHUB_AUTH_TOKEN env variable", func(t *testing.T) {
+		// Set environment variable
+		t.Setenv("GITHUB_AUTH_TOKEN", alternateText)
+		t.Setenv("GITHUB_TOKEN", "")
+		// Call your Load function
+		cfg := envconfig.Load()
+
+		// Assert expected value in the config
+		got, _ := cfg.GitHubToken.Get()
+		must.Eq(t, alternateText, got.String())
+	})
+
+	t.Run("loads from GITHUB_TOKEN when both GITHUB_TOKEN and GITHUB_AUTH_TOKEN env variable are set", func(t *testing.T) {
+		// Set environment variable
+		t.Setenv("GITHUB_AUTH_TOKEN", alternateText)
+		t.Setenv("GITHUB_TOKEN", text)
+
+		// Call your Load function
+		cfg := envconfig.Load()
+
+		// Assert expected value in the config
+		got, _ := cfg.GitHubToken.Get()
+		must.Eq(t, text, got.String())
+	})
+
+	t.Run("loads from the GITHUB_TOKEN env variable", func(t *testing.T) {
+		// Set environment variable
+		t.Setenv("GITHUB_TOKEN", text)
+
+		// Call your Load function
+		cfg := envconfig.Load()
+
+		// Assert expected value in the config
+		got, _ := cfg.GitHubToken.Get()
+		must.Eq(t, text, got.String())
+	})
+
+	//nolint:parallelTest // setting env vars
+
+	t.Run("returns none when no env is set", func(t *testing.T) {
+		t.Parallel()
+		// Set environment variable
+		os.Unsetenv("GITHUB_AUTH_TOKEN")
+		os.Unsetenv("GITHUB_TOKEN")
+
+		// Call your Load function
+		cfg := envconfig.Load()
+
+		// Assert expected value in the config
+		got := cfg.GitHubToken
+		must.Eq(t, prelude.None[configdomain.GitHubToken](), got)
+	})
+}
```

Comment:
This is a really nice test suite, and I'm perfectly happy to merge it the way it is right now. If you want to learn a nice trick, you could simplify these tests into a [table-driven test](https://go.dev/wiki/TableDrivenTests).

Reply:
I have updated the test to use the table driven tests. Kindly have a look 🙏